src/permutation.rs

```diff
@@ -37,22 +37,20 @@ impl Default for Permutation {
 }
 
 impl Permutation {
-    pub fn new(t: usize, full_rounds: usize, partial_rounds: usize) -> Result<Self, PermError> {
-        if full_rounds % 2 != 0 {
-            return Err(PermError::FullRoundsOdd);
-        }
-
-        let perm = Permutation {
+    // `full_rounds` input aims to be the number of full rounds that will be done
+    // before the partial rounds AND (BUT NOT ADDED TO) the number of full rounds 
+    // that will be done after the partial rounds.
+    pub fn new(t: usize, full_rounds: usize, partial_rounds: usize) -> Self {
+        
```

Comment:
Here, we should create a new variable called `total_num_full_rounds` to help differentiate between `full_rounds` and `total_num_full_rounds`

Reply:
As you told me on our latest conversation, it has been done as accorded. 

Hope on this way is fine now!